libs/qCC_glWindow/ccGLWindow.cpp

```diff
@@ -6212,6 +6212,13 @@ QString ccGLWindow::getShadersPath()
 		
 		shaderPath = (theDir.absolutePath() + "/share/cloudcompare/shaders");
 	}
+	else if ( theDir.dirName() == "qCC" ) // when running from build dir
```

Comment:
What's this 'qCC' folder? On which OS are you working? (Linux I guess?)

Reply:
I would also recommend that this be wrapped in `#ifdef QT_DEBUG`. No reason to have it active in a release version...

Reply:
Jup, I'm using linux. There it builds by default in <source_dir>/build/qCC/, so it can find the shaders in <sourcedir>/qCC/shaders. (It can build in a different location of course, so it might be better to let cmake copy the files to the build dir and use that, but that's way more hassle.)

QT_DEBUG is an option, but is not strictly related to where it is run from, you can test a release version too by running it in the build dir (actually that's what I do now). It shouldn't get to this if clause anyway, since any installed version will be installed in a 'bin' dir. Saying that I'd say it makes sense to remove the reasonable default shader dir in the else clause.

A similar issue exists with plugins, they are not loaded when running from the build dir, but that's a bit more difficult since every plugin is build in its own directory.

For Mac OS X it is done with the `CC_MAC_DEV_PATHS` variable, plus you have to copy (or symlink) the shaders dir into your build dir yourself.
But if things can be done automatically I'd say that's definitely preferred.

BTW in the Mac case the way to go up one dir seems to be done in a dangerous way: `appPath.remove("MacOS");`. If there is somewhere 'MacOS' in the path it gets removed! (for example if you put your CloudCompare.app in a dir called 'nice MacOS tools' on your USB disk it looks for plugins and shaders in the dir 'nice  tools' :/ ). Should better be done with a QDir like the linux case or with '../'.

How is it handled on Windows? Is it possible to run from the build dir there? Does it load shaders and plugins?

I still think this can be merged as is, but let me know.

Reply:
Actually I always use the INSTALL project. On Windows you have the issue that if you don't do that, in the best case you'll miss the necessary DLLs, or in the worst case wrong versions will be used instead!

Is there any reason why you don't use 'make install'?

Reply:
I like to keep a stable working version installed, so I can always run that, even if I'm testing some experimental feature in my build dir. That way I can better compare the experimental feature, and also I know I'll have a working version around when I need it. That way it's easier to spend some spare time on coding while not messing up the installed version.

In the case of shaders (which I'm now playing around with, see #584) it's even enough to just restart CloudCompare without recompile, which is way quicker. (I can find ways around it of course, and for now the easiest workaround is to have this commit in my feature branch and rebase it out in case of a PR)

The wrong version case can also exist in the Linux case because of the 'reasonable default' else clause which makes it look in /usr. Is there any reason to keep that? If you agree I can remove it in this PR too.

Reply:
Ok, I'll let @asmaloney and @rjanvier decide whether this can be merged in the main branch or not (I don't know enough about Linux!). And regarding the '/usr', same thing, I'm clueless :D

Reply:
to be paragmatic: if you only change he setup in a non "make install" case AND do not change anything to the "make install" case I guess i'm ok with this PR. if you change the behavior of the install case, I'm afraid I won't be agree because we follow the Filesystem Hierarchy standard.